app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue

```diff
@@ -51,13 +55,27 @@ const { floatingStyles } = useFloating(widgetRoot, dropdownElement, {
           mainAxis: (NODE_HEIGHT - state.rects.reference.height) / 2,
         }
       }),
-      size({
-        apply({ elements, rects }) {
-          Object.assign(elements.floating.style, {
-            minWidth: `${rects.reference.width + 16}px`,
-          })
+      size(() => ({
+        elementContext: 'reference',
+        apply({ elements, rects, availableWidth }) {
+          const PORT_PADDING_X = 8
+          const overflowing = Math.max(
+            (rects.floating.width - availableWidth) / 2 + PORT_PADDING_X,
+            0,
+          )
+          console.log('availableWidth', availableWidth)
+          const portWidth = rects.reference.width + PORT_PADDING_X * 2
+
+          const minWidth = `${Math.max(portWidth - overflowing, 0)}px`
```

Comment:
This line confuses me. If this is the minimum width of drop down, why do we _subtract_ overflowing from port width? The more the drop-down overflow, the less minimum width is there? Or I misread something?

Reply:
The more the "port width" overflows the screen borders, the less minimum width we want to enforce for dropdown. That way the dropdown will not needlessly stay large when it doesn't have to while the port is partially out of screen.

Reply:
Ok. So the important information missing for me is: overflow is actually "screen-overflow", I thought it's rather "port width overflow" (availableWidth is quite vague - I understood it as "availableWith in reference element"). You may adjust names a bit to make it clearer

Reply:
The `availableWidth` is unfortunately a property name defined by floating UI library. It looks like it is vague on purpose, since its actual meaning depends on many options. I'll just rename the local variable.